### ISSUE TITLE
Unsupported cases for `aten.amin/amax` conversion

### ISSUE THREAD

**Issue by jerrysky3** [tenstorrent/tt-metal#12936]
When trying to convert `aten.amin/amax` to `ttnn.min/max`, we ran into the issues below with tt-metal and currently don't support them:

- [ ] `keepdim = false` isn't supported in tt-metal
- [ ] Reduction on dims < rank - 2 results in assertion failure `Unsupported dim` inside tt-metal
- [ ] Rank of input shape < 2 gives `(1, 1)` output shape instead of the expected `(1)`
- [ ] `(..., x, y)` input shape where `x or y % ttnn.TILE_SIZE != 0`. In this case `TILE_LAYOUT` pads the input tensor with 0 which gives out incorrect results in min/max operation. It should pads `inf/-inf` corresponding to `ttnn.min/max`
- [ ] When the output shape is `(x, ttnn.TILE_SIZE * y)` where `x % ttnn.TILE_SIZE != 0`, tt-metal tries to reshape the padded output back to the expected output shape and results in assertion failure `Unable to reshape a tensor in TILE_LAYOUT to non-tile height and width! Please convert the tensor to ROW_MAJOR_LAYOUT first.`

**Comment by jerrysky3** [comment]
With a little more tests, we also found `ttnn.min/max(input_tensor)` always return 0 even if all values < 0

**Comment by jerrysky3** [comment]
> [@jerrysky3](https://github.com/jerrysky3) could you please provide one or more concrete examples of calling ttnn.max for the last case ("When the output shape is (x, ttnn.TILE_SIZE * y) where x % ttnn.TILE_SIZE != 0")?
> 
> Do you mean something like
> 
> ```
> def test_max3(device): 
>     x = -torch.ones((3, 1, 3, 64), dtype=torch.float32)
>     x_tt = ttnn.from_torch(x, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
>     keepdim = False # True
>     y_tt = ttnn.max(x_tt, dim=0, keepdim=keepdim)
>     y = ttnn.to_torch(y_tt)
>     z = torch.max(x, dim=0, keepdim=keepdim).values
>     print(f'x {x}\ny_tt {y_tt}\nz {z}')
>     assert_with_pcc(z, y)
> ```

Hi @bbradelTT , thanks for looking into this. I retested the original test cases in #241 and seems like it is no longer reprodcible:

```py
import ttnn
import torch


def main(device):
    x = torch.rand((1, 32), dtype=torch.bfloat16); dim = 1
    # x = torch.rand((4, 32), dtype=torch.bfloat16); dim = 0
    x_tt = ttnn.from_torch(x, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
    keepdim = True
    y_tt = ttnn.max(x_tt, dim=dim, keepdim=keepdim)
    y = ttnn.to_torch(y_tt)
    z = torch.max(x, dim=dim, keepdim=keepdim).values
    print(f"x {x}\ny_tt {y_tt}\nz {z}")
    print(torch.allclose(y, z))


if __name__ == "__main__":
    try:
        device = ttnn.open_device(device_id=0)
        main(device)
    finally:
        ttnn.close_device(device)

```

**Comment by bbradelTT** [comment]
Thanks @jerrysky3

**Comment by bbradelTT** [comment]
@prajaramanTT yes this is still an open issue.

**Comment by bbradelTT** [comment]
All of these should work now. Closing.

Verified the input shape one via:
```
import pytest
import torch
import ttnn
from tests.ttnn.utils_for_testing import assert_with_pcc

def test_max2(device):
    x = -torch.ones((1, 1, 3, 4), dtype=torch.float32)
    x_tt = ttnn.from_torch(x, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
    keepdim = False
    y_tt = ttnn.max(x_tt, dim=None, keepdim=keepdim)
    print(y_tt.shape)
    y = ttnn.to_torch(y_tt)
    z = torch.max(x)
    print(f'y {y}\nz {z}')
    assert_with_pcc(z, y)

def test_max3(device):
    x = torch.ones((4), dtype=torch.float32)
    x_tt = ttnn.from_torch(x, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
    keepdim = False
    y_tt = ttnn.max(x_tt, dim=None, keepdim=keepdim)
    print(y_tt.shape)
    y = ttnn.to_torch(y_tt)
    z = torch.max(x)
    print(f'y {y}\nz {z}')
    assert_with_pcc(z, y)

def test_max3(device):
    x = torch.ones((4), dtype=torch.float32)
    x_tt = ttnn.from_torch(x, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
    keepdim = True
    y_tt = ttnn.max(x_tt, dim=-1, keepdim=keepdim)
    print(y_tt.shape)
    y = ttnn.to_torch(y_tt)
    z = torch.max(x, -1, keepdim=keepdim).values
    print(f'y {y}\nz {z}')
    assert_with_pcc(z, y)
```
